File: app/frontend/modules/trigger/domain/trigger-repository.ts
import { Trigger } from './trigger'

export interface TriggerRepository {
  findId(uuid: string): Promise<Trigger>
  findByDiscordServerId(discordServerId: string): Promise<Trigger[]>
  save(trigger: Trigger): Promise<void>
  delete(trigger: Trigger): Promise<void>
}

File: context/src/main/kotlin/com/usadapekora/context/trigger/domain/TriggerRepository.kt
package com.usadapekora.context.trigger.domain

interface TriggerRepository {
    fun all(): Array<Trigger>
    fun find(id: Trigger.TriggerId): Trigger
    fun findByDiscordServer(id: Trigger.TriggerDiscordGuildId): Array<Trigger>
    fun save(trigger: Trigger)
    fun delete(trigger: Trigger)
}
